#!/usr/bin/env python
PACKAGE = "velodyne_detect_person"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("resolution",		double_t,    	0, "Resolution of Octree differ", 0.5,	0.1,   4)
gen.add("cluster_tolerance", 	double_t,	0, "Cluster Tolerance",   	  1.0,	0,   5)
gen.add("min_cluster_size",	int_t,		0, "Min Cluster Size of Cluster", 20,	0, 200)


# gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                       gen.const("Medium",     int_t, 1, "A medium constant"),
#                       gen.const("Large",      int_t, 2, "A large constant"),
#                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")], "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)




exit(gen.generate(PACKAGE, "velodyne_detect_person", "ExtractClusters"))
